Compile bug: error: passing 'const ggml_fp16_t *' (aka 'const unsigned short *') to parameter of type 'ggml_fp16_t *' (aka 'unsigned short *') discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]

### Git commit

4381

### Operating systems

BSD

### GGML backends

CPU

### Problem description & steps to reproduce

Build fails.

Triggered by: ci/run.sh

### First Bad Commit

_No response_

### Relevant log output

```shell
/usr/ports/misc/llama-cpp/work/llama.cpp-b4381/ggml/src/ggml-cpu/ggml-cpu.c:1602:39: error: passing 'const ggml_fp16_t *' (aka 'const unsigned short *') to parameter of type 'ggml_fp16_t *' (aka 'unsigned short *') discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
 1602 |             ax[j] = GGML_F16_VEC_LOAD(x + i + j*GGML_F16_EPR, j);
      |                     ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/misc/llama-cpp/work/llama.cpp-b4381/ggml/src/ggml-cpu/ggml-cpu.c:1024:55: note: expanded from macro 'GGML_F16_VEC_LOAD'
 1024 | #define GGML_F16_VEC_LOAD(p, i)      GGML_F32Cx4_LOAD(p)
      |                                      ~~~~~~~~~~~~~~~~~^~
/usr/ports/misc/llama-cpp/work/llama.cpp-b4381/ggml/src/ggml-cpu/ggml-cpu.c:1014:50: note: expanded from macro 'GGML_F32Cx4_LOAD'
 1014 | #define GGML_F32Cx4_LOAD(x)     __sse_f16x4_load(x)
      |                                                  ^
/usr/ports/misc/llama-cpp/work/llama.cpp-b4381/ggml/src/ggml-cpu/ggml-cpu.c:989:52: note: passing argument to parameter 'x' here
  989 | static inline __m128 __sse_f16x4_load(ggml_fp16_t *x) {
      |                                                    ^
2 errors generated.
```
```

--- a/ggml/src/ggml-cpu/ggml-cpu.c
+++ b/ggml/src/ggml-cpu/ggml-cpu.c
@@ -986,7 +986,7 @@ inline static void __wasm_f16x4_store(ggml_fp16_t * p, v128_t x) {
 #define GGML_F16_STEP 32
 #define GGML_F16_EPR  4
 
-static inline __m128 __sse_f16x4_load(ggml_fp16_t *x) {
+static inline __m128 __sse_f16x4_load(const ggml_fp16_t * x) {
     float tmp[4];
 
     tmp[0] = GGML_FP16_TO_FP32(x[0]);
@@ -997,7 +997,7 @@ static inline __m128 __sse_f16x4_load(ggml_fp16_t *x) {
     return _mm_loadu_ps(tmp);
 }
 
-static inline void __sse_f16x4_store(ggml_fp16_t *x, __m128 y) {
+static inline void __sse_f16x4_store(ggml_fp16_t * x, __m128 y) {
     float arr[4];
 
     _mm_storeu_ps(arr, y);

